doc/dev/adr/0023-large-message-chunking.md

```diff
@@ -0,0 +1,120 @@
+﻿# ADR 23: Large Message Chunking in MQTT Protocol
+
+## Status
+
+Proposed
+
+## Context
+
+The MQTT protocol has inherent message size limitations imposed by brokers and network constraints. Azure IoT Operations scenarios often require transmitting payloads that exceed these limits (e.g., firmware updates, large telemetry batches, complex configurations). Without a standardized chunking mechanism, applications must implement their own fragmentation strategies, leading to inconsistent implementations and interoperability issues.
+
+## Decision
+
+We will implement sdk-level message chunking as part of the Protocol layer to transparently handle messages exceeding the MQTT broker's maximum packet size.
+
+**The chunking mechanism will**:
+
+- Be enabled/disabled by a configuration setting.
+- Use standardized user properties for chunk metadata. The `__chunk` user property will contain a colon-separated string with chunking metadata: ```<messageId>:<chunkIndex>:<totalChunks>:<checksum>```. The string will include:
+  - `messageId` - UUID string in the 8-4-4-4-12 format, present for every chunk.
+  - `chunkIndex` - unsigned 32 bit integer in decimal format, present for every chunk;
+  - `totalChunks` - unsigned 32 bit integer in decimal format, present only for the first chunk.
+  - `checksum` - SHA-256 hash in hexadecimal format (64 characters long), present only for the first chunk.
+
+**Chunk size calculation:**
+
+- Maximum chunk size will be derived from the MQTT CONNECT packet's Maximum Packet Size.
```

Comment:
* I like having the maximum size based off CONNACK.  
  * Lower prio - it maybe OK to punt in v1 - but you should think whether you want this to also be configured via a knob, where the size would be min(CONNACK calculation, knob setting).
  * The scenario is that we could imagine say a low-end device that only had some small amount of RAM, where we only want to send is smaller chunks even if the MQTT broker allowed larger sizes.